build.gradle

```diff
@@ -1,3 +1,103 @@
+import groovy.json.JsonSlurper
+
 group 'com.amazonaws.blox'
 version '0.1-SNAPSHOT'
 description "Blox: Open Source schedulers for Amazon ECS"
+
+buildscript {
+    repositories {
+        maven { url 'https://plugins.gradle.org/m2/' }
+    }
+
+    dependencies {
+        classpath 'com.diffplug.gradle.spotless:spotless:2.4.1'
+    }
+}
+
+def unformattedProjects = [
+        'frontend-infrastructure',
+        'frontend-service-client'
+]
+
+allprojects {
+    apply plugin: 'com.diffplug.gradle.spotless'
+}
+
+configure(subprojects.findAll { !unformattedProjects.contains(it.name) }) {
+    spotless {
+        java {
+            googleJavaFormat()
+            licenseHeaderFile rootProject.file('licenses/license-header.java')
+        }
+    }
+}
+
+ext {
+    // This can be overridden by specifying -PawsProfile=my-other-profile
+    // on the Gradle command line:
+    awsProfile = project.hasProperty("awsProfile") ? project.awsProfile : "default"
+    println("Profile: ${awsProfile}")
+
+    awsCli = "/usr/local/bin/aws"
+    awsRegion = "us-west-2"
+    awsPrefix = System.getenv("USER")
+
+    stackName = "${awsPrefix}-blox-frontend-${awsRegion}"
+    s3BucketName = "${stackName}-${awsProfile}"
+    stageName = "Beta"
```

Comment:
Can we quickly whiteboard out how we want to make this configurable? As mentioned in another comment, all this AWS settings are just super draft to make it work.